### PR TITLE
Add support for IOCTL

### PR DESCRIPTION
**Pull request by steinbeck65**
ioctl.py - example code for handling IOCTL requests
ioctl.c - example code for sending IOCTL requests

### CODE REVIEW

**Comment by eestrada** [comment]
@steinbeck65 This looks good to me. Before merging this, will you first move `ioctl.py` and `iotctl.c` to live under `examples`? I am trying to keep the root directory as minimal as possible.

**Comment by steinbeck65** [comment]
I've moved the ioctl.c and ioctl.py files to the examples directory and
squashed those changes into the original commit, hopefully I did that all
correctly. Let me know if I can do anything else.

Thank you very much for breathing new life into fusepy, I and I'm sure many
other really appreciate it!

--
Greg

**Comment by eestrada** [comment]
Thanks for taking care of that. The squashing looks like it worked correctly.

You're welcome! `fusepy` is an important project that needs to remain alive. I'm happy to do what I can to support it.

File: examples/ioctl.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env python
+# Example code for handling IOCTLs in FUSE client
+
+import ctypes
+import struct
+from errno import ENOTTY
+from ioctl_opt import IOWR
+
+def ioctl(self, path, cmd, arg, fh, flags, data):
```
Comment (terencehonles) [inline examples/ioctl.py]: Can you define the whole operations class? As it stands this example "does not work"

**Comment by steinbeck65** [comment]
I created a minimal fuse file system using memory.py as a starting point,
then added in the ioctl support, so it now works as a standalone example.

I've squashed these changes into the original pull request, let me know if
that is satisfactory or if you would like any additional changes.

Thanks

--
Greg

**Pull request merged**